### ISSUE TITLE
Feature request: %debug to jump to last user code in Traceback

### ISSUE THREAD

**Issue by gerazov** [spyder-ide/spyder#15216]
Debugging is one of the strongest points of Spyder. It's awesome how you can use the `%debug` magic to jump to the last line in the error traceback and explore the variables.

Unfortunately, if the last file was an imported module then it's not as informative, and it gets less informative the more nested the imported functions used. In this case, it would be useful if `%debug` jumped back to the last user file.

**Comment by ccordoba12** [comment]
Hey @gerazov, that's a really good idea!

@impact27, what do you think about this?

**Comment by impact27** [comment]
I agree with this, but this should only be the case if Preferences -> Ipython console -> Debugger -> Ignore Python libraries while debugging is on

**Comment by ccordoba12** [comment]
Agreed.

**Comment by gerazov** [comment]
Makes sense - that way you could have it either way :+1:
